src/conversions.jl

```diff
@@ -597,8 +597,16 @@ end
 function PyObject(d::AbstractDict)
     o = PyObject(@pycheckn ccall((@pysym :PyDict_New), PyPtr, ()))
     for k in keys(d)
+        pykey = PyObject(k)
+        pyval = PyObject(d[k])
```

Comment:
Would be cleaner to have a function
```
function valid_pyobject(x)
    o = PyObject(x)
    ispynull(o) && error("invalid PyNULL object created from $x")
    return o
end
```
and call that in cases where we want to check for NULL results.